### PR TITLE
Add apidoc

### PR DESCRIPTION
**Pull request by dangunter**
## Fixes/Addresses: 

Adding auto-generated API docs back in to the repo. Including instructions for how to generate them.

Question: do we need to check the ``apidoc/*.rst`` files in to Git?

## Summary/Motivation:

If the auto-generated API doc files are there, references in code docstrings between pieces should be navigable. Also, it's just nice to have.

## Changes proposed in this PR:
- add commands for Windows and Linux/OSX to generate API docs with correct options 
- add developer documentation
- add appropriate TOC entries for the generated API docs

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by lbianchi-lbl — CHANGES_REQUESTED** [review]
I've checked the built apidocs on ReadTheDocs, and I think they look awesome. The accompanying developer docs also look good. I left a couple of questions and comments, mostly on test robustness.

**Review by TimBartholomew — DISMISSED** [review]

**Comment by lbianchi-lbl** [comment]
I'll keep an eye on this after #223 is merged to see how the test behaves when no docs directory is found, and possibly just add a `pyest.skip()` if needed.

File: docs/getting_started/install.rst [hunk 1]
```diff
@@ -150,8 +152,76 @@ When either the ``proteuslib`` package or one of its dependencies are installed,
...
+To generate a local copy of the documentation for the first time, follow these steps:
+
+1. Change directory to the "docs" subdirectory
+
+2. Generate the tree of API documentation with "sphinx-apidoc". For convenience, a script has been
+   provided that has all the required options.
+
+   * On Windows, run ``.\apidoc.bat``
+
+   * On Linux/OSX run ``./apidoc.sh``
```
[the remark this answers is not in the document]
Comment (dangunter) [reply docs/getting_started/install.rst:255]: We could add that too, but (a) make/gmake is not as ubiquitous in Windows as Mac/Linux, (b) modifying Sphinx Makefiles is a little less obvious than these extremely simple scripts.

File: proteuslib/tests/test_docs.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+"""
+Tests for documentation
+"""
+from pathlib import Path
+import re
+
+__authors__ = ["Dan Gunter"]
+
+
```
[the remark this answers is not in the document]
Comment (dangunter) [reply proteuslib/tests/test_docs.py:9]: Fine to expose this more formally, but the sphinx check seems like added complexity to me.

This is a little worrisome for reasons you don't describe, i.e. can I mess with formatting in a way to "fool" it? Maybe.

**Comment by dangunter** [comment]
@lbianchi-lbl I added skip. I think it would have worked anyways, but now it does everything except check for Sphinx. That still strikes me as fragile, who knows why it might not be in the path.

**Review by TimBartholomew — APPROVED** [review]

**Pull request merged**